src/pages/EditMovie.js

```diff
@@ -1,26 +1,50 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Redirect } from 'react-router-dom';
 
-import { MovieForm } from '../components';
+import { MovieForm, Loading } from '../components';
 import * as movieAPI from '../services/movieAPI';
 
 class EditMovie extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      status: 'loading',
+      movie: {},
+      shouldRedirect: false,
+    };
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
+  async componentDidMount() {
+    this.getMovieAPI();
+  }
+
+  async getMovieAPI() {
+    const { id } = this.props.match.params; // pegando o id da URL
+    const movieResult = await movieAPI.getMovie(id);
+    this.setState({ movie: movieResult, status: '' });
+  }
+
   handleSubmit(updatedMovie) {
+    console.log(updatedMovie);
+    movieAPI.updateMovie(updatedMovie).then((response) => { // para pegar resposta do update movies
+      if (response === 'OK') {
+        this.setState({ shouldRedirect: true });
+      }
+    });
   }
```

Comment:
Ju, esse if na linha 32 não fez muito sentido pra mim. Como você usou o .then, esse próprio método vai realizar sua tarefa apenas se a resposta for positiva (no caso se se response === 'OK'). No caso dessa função que eles disponibilizaram pra gente (movieAPI.updateMovie), não tem nem como ela falhar! Porque ela já foi projetada pra esperar um tempo e retornar apenas o sucesso, não tem nenhuma possibilidade de 'reject' nela. Uma outra sugestão de como fazer seria:
```suggestion
  handleSubmit(updatedMovie) {
    movieAPI
      .updateMovie(updatedMovie)
      .then(() => this.setState({ shouldRedirect: true }));
      // no .then, como a gente não vai usar a 'response' (que é simplesmente um 'OK'), não precisa nem colocar ela no retorno da função, apenas usar o setState mesmo
  }
```
Ou com async/await:
```suggestion
  async handleSubmit(updatedMovie) {
    await movieAPI.updateMovie(updatedMovie);
    await this.setState({ shouldRedirect: true });
  }
```